### PR TITLE
CRUD operations added.

### PR DESCRIPTION
**Pull request by jlandwehr-avarteq**
On this branch, all CRUD operations were added. Furthermore, a database entry for articles was created and the application was extended by different views that show the the implemented functionality. 

@arkanoryn

### CODE REVIEW

**Comment by arkanoryn** [comment]
Hello @jlandwehr-avarteq 

Your initial message is not clear enough IMO. Just like that, I have no idea what your exercise was.

Please, try to something like:

# Chapter X - link_to_chapter

## previous state

Before this request, I only had a simple rails app

## aimed state
- I aim to create CRUD operation on model `model_name` and their related views.
- Create database table for the new model `model_name`

File: blog/app/controllers/articles_controller.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+		@article = Article.find(params[:id])
+		@article.destroy
+		redirect_to articles_path
+	end 
+
+	private
+	def article_params
+		params.require(:article).permit(:title, :text)
+	end
+
```
Comment (arkanoryn) [inline blog/app/controllers/articles_controller.rb]: At Avarteq we remove the empty line. :)

File: blog/app/models/article.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+class Article < ActiveRecord::Base
+	validates :title, presence: true,
+		length: {minimum: 5}
```
Comment (arkanoryn) [inline blog/app/models/article.rb]: usually, validations stay on one line and get cut only if the `character by line` limit is met.

File: blog/app/views/articles/edit.html.erb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<h1>Editing article</h1>
+ 
+<%= render 'form' %>
+ 
+<%= link_to 'Back', articles_path %>
+
```
Comment (arkanoryn) [inline blog/app/views/articles/edit.html.erb]: empty lines?

File: blog/app/views/articles/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    	<th>Text</th>
+    	<th colspan="2"></th>
+  	</tr>
+	<% @articles.each do |article| %>
+		<tr>
+      		<td><%= article.title %></td>
+      		<td><%= article.text %></td>
+      		<td><%= link_to 'Show', article_path(article) %></td>
+      		<td><%= link_to 'Edit', edit_article_path(article) %></td>
+      		<td><%= link_to 'Delete', article_path(article),
```
Comment (arkanoryn) [inline blog/app/views/articles/index.html.erb]: if you go for multiline, do multiline:

```
<%= link_to 'Delete', 
article_path(article), 
method: :delete, 
data: {confirm: 'Are you sure?'} %>
```

I let you do the alignment.

File: blog/config/routes.rb [hunk 5]
```diff
@@ -1,5 +1,6 @@
 Rails.application.routes.draw do
-  get 'welcome/index'
+ 
+  resources :articles
```
Comment (arkanoryn) [inline blog/config/routes.rb]: I personally prefer to keep `root` at the top of the `Routes` file.

This way, when someone open the file, he directly knows where it starts.

Try to keep things organized as much as possible. ;)

File: blog/app/views/articles/show.html.erb [hunk 6]
```diff
@@ -0,0 +1,10 @@
+<p>
+	<strong>Title:</strong>
+  	<%= @article.title %>
+</p>
+<p>
+	<strong>Test:</strong>
+	<%= @article.text %>
```
Comment (arkanoryn) [inline blog/app/views/articles/show.html.erb]: Get used to use the function `simple_format` on view having a text field. 
http://apidock.com/rails/ActionView/Helpers/TextHelper/simple_format

_database textfield != string_

**Review by arkanoryn — COMMENTED** [review]
Hello @jlandwehr-avarteq 

A few comments to fix before going further.
let me know when it is fixed.

Before linking me, please rebase on top of master.
As said before, we can look into it this afternoon.

**Comment by jlandwehr-avarteq** [comment]
Wrong branch target.

**Pull request closed without merging**